### PR TITLE
[FEATURE] Introduce php case and backed enums

### PR DESCRIPTION
**Pull request by linawolf**

### CODE REVIEW

File: tests/integration/enum-backed-with-cases-warning/input/index.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
+===================
+PHP Enum with Const
```
Comment (brotkrueml) [inline tests/integration/enum-backed-with-cases-warning/input/index.rst]: Heading does not match purpose (like given in the folder name)

File: tests/integration/enum-backed-with-cases-warning/input/index.rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+===================
+
+..  php:enum:: RedSuit : string
```
Comment (brotkrueml) [inline tests/integration/enum-backed-with-cases-warning/input/index.rst]: Folder has "with cases warning" - but I assume, there will be also an enum warning (string defined with `: string` and with `:type:`

So the folder name (and therefore the heading) should be adjusted. Or a separate test should be used.

File: tests/integration/enum-backed-with-cases/input/index.rst [hunk 3]
```diff
@@ -0,0 +1,19 @@
+===================
+PHP Enum with Const
```
Comment (brotkrueml) [inline tests/integration/enum-backed-with-cases/input/index.rst]: Here also wrong heading.

**Review by brotkrueml — CHANGES_REQUESTED** [review]

File: tests/integration/enum-backed-with-cases-warning/input/index.rst (hunk 2 above)
Comment (linawolf) [reply tests/integration/enum-backed-with-cases-warning/input/index.rst]: renamed both folder and headline

File: tests/integration/enum-backed-with-cases/input/index.rst (hunk 3 above)
Comment (linawolf) [reply tests/integration/enum-backed-with-cases/input/index.rst]: renamed headline

File: tests/integration/enum-backed-with-cases-warning/input/index.rst (hunk 1 above)
Comment (linawolf) [reply tests/integration/enum-backed-with-cases-warning/input/index.rst]: renamed

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**